sklearn/metrics/classification.py

```diff
@@ -1225,14 +1250,45 @@ class 2       1.00      0.67      0.80         3
     else:
         labels = np.asarray(labels)
 
-    last_line_heading = 'avg / total'
+    is_binary_classification = len(labels) == 2
+    is_multilabel = _check_targets(y_true, y_pred)[0].startswith('multilabel')
+
+    if is_binary_classification and average:
+        raise ValueError('Averaging methods are not relevant for binary '
+                         'classification. Please omit average argument.')
+
+    if average:
+        if isinstance(average, basestring):
+            averages = (average,)
+        else:
+            averages = average
+        average_options = (None, 'micro', 'macro', 'weighted', 'samples')
+        if not set(averages).issubset(average_options):
+            raise ValueError('average has to be one or more of ' +
+                             str(average_options))
+        if 'samples' in averages and not is_multilabel:
+            raise ValueError("'samples' averaging method is only allowed "
+                             "for multi-label classifications.")
+    else:
+        if is_binary_classification:
+            averages = (None,)  # dummy entry to let totals show up
+        elif is_multilabel:
+            averages = ('samples', 'weighted', 'micro', 'macro')
+        else:
+            averages = ('weighted', 'micro', 'macro')
+
+    if not is_binary_classification:
+        last_line_headings = [a + ' avg / total' for a in averages]
+    else:
+        last_line_headings = ['total']
```

Comment:
It's useful to display the total number of samples even for the binary case. averaging precision / recall / f1 score for binary does not make sense for binary classification though (as discussed in the next comment).